Tests/Functional/UpsertFunctionalTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace MauticPlugin\CustomObjectsBundle\Tests\Functional;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Mautic\LeadBundle\Provider\FilterOperatorProviderInterface;
+use MauticPlugin\CustomObjectsBundle\CustomFieldType\TextType;
+use MauticPlugin\CustomObjectsBundle\Entity\CustomField;
+use MauticPlugin\CustomObjectsBundle\Entity\CustomItem;
+use MauticPlugin\CustomObjectsBundle\Entity\CustomObject;
+use MauticPlugin\CustomObjectsBundle\Tests\Functional\DataFixtures\Traits\CustomObjectsTrait;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class UpsertFunctionalTest extends \Mautic\CoreBundle\Test\MauticMysqlTestCase
+{
+    use CustomObjectsTrait;
+
+    private CustomItem $existingCustomItem;
+    private ?object $customItemModel;
+    private CustomObject $customObject;
+    private ?object $customItemRepository;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->customItemModel       = self::$container->get('mautic.custom.model.item');
+        $this->customItemRepository  = self::$container->get('custom_item.repository');
+        $this->customObject          = $this->createCustomObject();
+        $this->existingCustomItem    = $this->createCustomItem($this->customObject, 'Sapiens');
+        $this->existingCustomItem->createNewCustomFieldValueByFieldAlias('unique_id_field', 'SomeUniqueHash');
+        $this->existingCustomItem = $this->customItemModel->save($this->existingCustomItem);
+    }
+
+    public function testCreateNewCustomItemWithNewUniqueHash(): void
+    {
+        $this->assertEquals(1, $this->customItemRepository->getItemCount($this->customObject->getId()));
+        $this->assertEquals('Sapiens', $this->existingCustomItem->getName());
+
+        $newCustomItem = $this->createCustomItem($this->customObject, 'Factfulness');
+        $newCustomItem->createNewCustomFieldValueByFieldAlias('unique_id_field', 'SomeOtherUniqueHash');
+        $newCustomItem = $this->customItemModel->save($newCustomItem);
+
+        $this->assertEquals(2, $this->customItemRepository->getItemCount($this->customObject->getId()));
+        $this->assertNotEquals($this->existingCustomItem->getUniqueHash(), $newCustomItem->getUniqueHash());
+        $this->assertEquals('Factfulness', $newCustomItem->getName());
+    }
```

Comment:
Isn't this test duplicated in the test bellow except the last line? Can the last line be added to the test bellow and this test deleted?

Reply:
Good call, thanks!